synapseclient/models/wiki.py

```diff
@@ -1 +1,804 @@
+
+        client = Synapse.get_client(synapse_client=synapse_client)
+
+        # Get the cache directory path to save the newly created gzipped file
+        cache_dir = os.path.join(client.cache.cache_root_dir, "wiki_markdown")
+        if not os.path.exists(cache_dir):
+            os.makedirs(cache_dir)
+
+        # Check if markdown looks like a file path and exists
+        if os.path.isfile(markdown):
+            # If it's already a gzipped file, save a copy to the cache
+            if markdown.endswith(".gz"):
+                file_path = os.path.join(cache_dir, os.path.basename(markdown))
+                shutil.copyfile(markdown, file_path)
+            else:
+                # If it's a regular html or markdown file, compress it
+                with open(markdown, "rb") as f_in:
+                    # Open the output gzip file
+                    file_path = os.path.join(
+                        cache_dir, os.path.basename(markdown) + ".gz"
+                    )
+                    with gzip.open(file_path, "wb") as f_out:
+                        f_out.writelines(f_in)
+
+        else:
+            # If it's a plain text, write it to a gzipped file and save it in the synapse cache
+            file_path = os.path.join(cache_dir, f"wiki_markdown_{uuid.uuid4()}.md.gz")
+            with gzip.open(file_path, "wt", encoding="utf-8") as f_out:
+                f_out.write(markdown)
+
+        return file_path
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Create_Wiki_Page: {self.owner_id}"
+    )
+    async def create_async(
+        self,
+        *,
+        synapse_client: Optional[Synapse] = None,
+        force_version: bool = False,
+    ) -> "WikiPage":
+        """Create a new wiki page.
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                    `Synapse.allow_client_caching(False)` this will use the last created
+                    instance from the Synapse class constructor.
+            force_version: If True, the wiki page will be created with a new version number.
+
+        Returns:
+            The created wiki page.
+
+        Raises:
+            ValueError: If owner_id is not provided or if required fields are missing.
+        """
+        if not self.owner_id:
+            raise ValueError("Must provide owner_id to create a wiki page.")
+        # check if the attachments exists
+        if not self.markdown:
+            raise ValueError("Must provide markdown content to create a wiki page.")
+
+        # Convert markdown to gzipped file if needed
+        file_path = self._markdown_to_gzip_file(self.markdown, synapse_client)
+
+        # Upload the gzipped file to get a file handle
+        file_handle = await upload_file_handle(
+            syn=synapse_client,
+            parent_entity_id=self.owner_id,
+            path=file_path,
+        )
+
+        # delete the temp gzip file
+        os.remove(file_path)
```

Comment:
This should be in a `try/finally` block to ensure that `os.remove` gets called even if an exception occurs